src/ParseFile.js

```diff
@@ -311,10 +314,16 @@ const DefaultController = {
   },
 
   download: function(uri) {
+    if (XHR) {
```

Comment:
I think I get it.  Fine as is.

re-writting below how I would formultate.  You think this is functionally the same?

no need to change, just curious.
```
if(process.env.PARSE_BUILD === 'browser') {
  if(!XHR) {
     return Promise.reject('Cannot make a request: No definition of XMLHttpRequest was found.');
  }

  return return this.downloadAjax(uri);
}
// We're in node
return new Promise((resolve, reject) => {
    const client = uri.indexOf('https') === 0 
        ? require('https')
        : require('http');
...
```

Reply:
Functionality will be the same

Reply:
Actually xmlhttprequest exists in react-native